### PR TITLE
If no keyEvents or availableTopics, don't show the filters or the filter heading in FilterTopicBank

### PR DESCRIPTION
**Pull request by sophie-macmillan**
## What does this change?
This change updates the check on availableTopics, to ensure that the array has some items in it. If not, we don't show the FilterTopic bank.

## Why?
UX Improvement - We've seen a case when availableTopics exists, but it empty, which meant the Filter headings were visible with no Filter buttons.

## Screenshots

| Before      | After      |
|-------------|------------|
| image| image|

### CODE REVIEW

File: dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx [hunk 1]
```diff
@@ -83,6 +83,9 @@ export const TopicFilterBank = ({
 	filterKeyEvents = false,
 	id,
 }: Props) => {
+	const showFilters = keyEvents?.length && availableTopics.length;
```
Comment (oliverlloyd) [inline dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx:86]: My brain melts in these situations so there's a non trivial chance I'm talking rubbish but should this be OR here or?

```suggestion
	const showFilters = keyEvents?.length || availableTopics.length;
```

Comment (sophie-macmillan) [reply dotcom-rendering/src/web/components/TopicFilterBank.importable.tsx]: Oops yeah, you're absolutely right!

**Review by mxdvl — APPROVED** [review]

**Review by oliverlloyd — APPROVED** [review]
👍

**Comment by sophie-macmillan** [comment]
I originally avoided this as the `TopicFilterBank` is used in both `LiveLayout.tsx` and `LiveBlogRenderer.tsx`, to reduce code duplication. If reducing javascript load is the priority, I could just duplicate this logic in both files?

**Comment by oliverlloyd** [comment]
Reducing javascript load

i-need-that-so-bad-marissa-rachel

**Review by OllysCoding — APPROVED** [review]
+1 Looks good!!

File: dotcom-rendering/src/web/layouts/LiveLayout.tsx [hunk 2]
```diff
@@ -750,7 +751,8 @@ export const LiveLayout = ({ CAPIArticle, NAV, format }: Props) => {
 								</Hide>
 
 								{showTopicFilterBank &&
-									CAPIArticle.availableTopics && (
+									CAPIArticle.availableTopics &&
```
Comment (bryophyta) [inline dotcom-rendering/src/web/layouts/LiveLayout.tsx:754]: if I'm understanding correctly then is looks like this set of 3 conditions is equivalent to `!showToggle`. If that's the intention, there might be scope to rename `showToggle` and then use it here as well as where it's used above.

Just a suggestion if I've understood correctly, and if it seems like it'll help to make the logic easier to follow. Not a blocking suggestion for this PR  though!

**Pull request merged**